app/src/main/java/com/zeapo/pwdstore/git/log/GitLogAdapter.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright © 2014-2020 The Android Password Store Authors. All Rights Reserved.
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+
+package com.zeapo.pwdstore.git.log
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.github.ajalt.timberkt.e
+import com.zeapo.pwdstore.databinding.GitLogRowLayoutBinding
+import java.text.DateFormat
+import java.util.Date
+
+private fun shortHash(hash: String): String {
+    return hash.substring(0 until 8)
+}
+
+private fun stringFrom(date: Date): String {
+    return DateFormat.getDateTimeInstance().format(date)
+}
+
+/**
+ * @see GitLogActivity
+ */
+class GitLogAdapter : RecyclerView.Adapter<GitLogAdapter.ViewHolder>() {
+
+    private val model = GitLogModel()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding = GitLogRowLayoutBinding.inflate(inflater, parent, false)
+        return ViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
+        val commit = model.get(position)
+        if (commit == null) {
+            e { "There is no git commit for view holder at position $position." }
+            return
+        }
+        viewHolder.bind(commit)
+    }
+
+    override fun getItemCount() = model.size
+
+    class ViewHolder(private val binding: GitLogRowLayoutBinding) : RecyclerView.ViewHolder(binding.root) {
+
+        fun bind(commit: GitCommit) {
+            binding.apply {
```

Comment:
```suggestion
        fun bind(commit: GitCommit) = with(binding) {
```

This reads a bit cleaner IMO